### DISCUSSION TITLE
feature: add f# extra

### DISCUSSION THREAD

**Discussion by liamwh** [LazyVim/LazyVim#3470] [Extra Requests]
### Did you check the docs?

- [X] I have read all the LazyVim docs

### Is your feature request related to a problem? Please describe.

Not sure how best to set up a Neovim environment for f# development with LazyVin

### Describe the solution you'd like

Would like an extra for f# development

### Describe alternatives you've considered

N/A

**Comment by dpetka2001** [comment]
PRs are welcome if you do f# yourself. Otherwise you'd have to wait for someone who does f# to submit one.

**Comment by muneebusmani** [comment]
i will take a look at that

**Comment by muneebusmani** [comment]
@liamwh use this plugin
```
return {
  "ionide/Ionide-vim",
  event = "LazyFile"
}
```

**Reply by liamwh** [reply]
Thanks for this! Should nothing else for f# be installed with Mason?

**Comment by muneebusmani** [comment]
just install fsautocomplete and fantomas (lsp and formatter respectively), its lsp has linting capabilities which are enabled by this plugin

**Reply by muneebusmani** [reply]
i think f#  needs an extra, i will implement it in an extra and submit it, but it is up to the maintainer to accept it or not

**Reply by liamwh** [reply]
Adding an extra would be amazing, thank you! 🙏🤩

**Reply by muneebusmani** [reply]
did it : #3564 , just wait for the maintainer's approval then it will be available in lazyvim
